types/parsimmon/index.d.ts

```diff
@@ -107,6 +107,13 @@ declare namespace Parsimmon {
          * about the error.
          */
         tryParse(input: string): T;
+        /**
+         * Passes the result of `parser` to the function `condition`,
+         * which returns a boolean. If the the condition is false, returns
+         * a failed parse with the given `message`. Else it returns the
+         * original result of `parser`.
+         */
+        assert<U>(condition: (result: U) => boolean, message: string): Parser<U>;
```

Comment:
Type `T` should be enough for the result.

see: https://github.com/jneen/parsimmon/blob/99af8f8d8dd0c1aa8542e8ef930805b295194086/src/parsimmon.js#L893-L897

Reply:
I've updated the PR to just use `T` :)

Reply:
Thank you!